packages/core/src/driver.ts

```diff
@@ -288,6 +382,106 @@ class Driver {
     this._afterConstruction()
   }
 
+  /**
+   * The bookmark managed used by {@link Driver.executeQuery}
+   *
+   * @experimental
+   * @type {BookmarkManager}
+   * @returns {BookmarkManager}
+   */
+  get queryBookmarkManager (): BookmarkManager {
+    return this._queryBookmarkManager
+  }
+
+  /**
+   * Executes a query in a retriable context and returns a {@link EagerResult}.
+   *
+   * This method is a shortcut for a transaction function.
+   *
+   * @example
+   * // Run a simple write query
+   * const { keys, records, summary } = await driver.executeQuery('CREATE (p:Person{ name: $name }) RETURN p', { name: 'Person1'})
+   *
+   * @example
+   * // Run a read query
+   * const { keys, records, summary } = await driver.executeQuery(
+   *    'MATCH (p:Person{ name: $name }) RETURN p',
+   *    { name: 'Person1'},
+   *    { routing: neo4j.routing.READERS})
+   *
+   * @example
+   * // Run a read query return a Person Node
+   * const person1 = await driver.executeQuery(
+   *    'MATCH (p:Person{ name: $name }) RETURN p',
+   *    { name: 'Person1'},
+   *    {
+   *      resultTransformer: neo4j.resultTransformers.mappedResultTransformer({
+   *        map(record) {
+   *          return record.get('p')
+   *        },
+   *        collect(personArray) {
+   *          return personArray[0]
+   *        }
+   *      })
+   *    }
+   * )
+   *
+   * @example
+   * // these lines
+   * const transformedResult = await driver.executeQuery(
+   *    "<QUERY>",
+   *    <PARAMETERS>,
+   *    {
+   *       routing: neo4j.routing.WRITERS,
+   *       resultTransformer: transformer,
+   *       database: "<DATABASE>",
+   *       impersonatedUser: "<USER>",
+   *       bookmarkManager: bookmarkManager
+   *    })
+   * // are equivalent to those
+   * const session = driver.session({
+   *    database: "<DATABASE>",
+   *    impersonatedUser: "<USER>",
+   *    bookmarkManager: bookmarkManager
+   * })
+   *
+   * try {
+   *    const transformedResult = await session.executeWrite(tx => {
+   *        const result = tx.run("<QUERY>", <PARAMETERS>)
+   *        return transformer(result)
+   *    })
+   * } finally {
+   *    await session.close()
+   * }
+   *
+   * @public
+   * @experimental
```

Comment:
will this annotation be enough to tell users this API can change anytime? or is an explicit mention worth it?